lib/1.4/dml-builtins.dml

```diff
@@ -2520,6 +2520,28 @@ template bank is (object, shown_desc) {
         }
     }
 
+    implement register_view_index {
+        method register_names() -> (attr_value_t) {
+            local (const register *table, uint64 table_size) = _reginfo_table();
+            local attr_value_t ret = SIM_alloc_attr_list(table_size);
+            for (local int i = 0; i < table_size; i++) {
+                SIM_attr_list_set_item(
+                    &ret, i, SIM_make_attr_string(
+                        table[i]._qname() + strlen(bank._qname()) + 1));
+            }
+            return ret;
+        }
+        method register_offsets() -> (attr_value_t) {
+            local (const register *table, uint64 table_size) = _reginfo_table();
+            local attr_value_t ret = SIM_alloc_attr_list(table_size);
+            for (local int i = 0; i < table_size; i++) {
+                SIM_attr_list_set_item(
+                    &ret, i, SIM_make_attr_uint64(table[i].offset));
+            }
+            return ret;
+        }
```

Comment:
Maybe we could construct these `attr_value_t`s in an independent startup memoized, and then have the implementations of these methods be a simple `SIM_attr_copy` on the relevant one fetched?

Reply:
Once we have RAII strings so we can get an independent `qname()` that _would_ be possible (it is possible even today but not without changes in dmllib.h). Do you think the approach I mentioned is a good direction to take? If so, add a TODO (and possibly a JIRA) for it. If not, then you don't need to bother.

Reply:
I don't see a big value in memoizing:
* current impl is probably not much slower than SIM_attr_copy
* memoizing is likely a net loss, assuming that in most simulations most banks are never register_view:ed.
* We can assume bottlenecks to be elsewhere until proven otherwise. 

What we *could* do is to recycle the `strlen(bank.qname)`. I'll do that.